src/js/modules/RoutingApp.js

```diff
@@ -110,20 +124,58 @@ class RoutingApp extends Component {
               <Route
                 exact
                 path={"/myaccount"}
-                key={"myaccount"}
-                component={SignInPage}
+                key={"signIn"}
+                render={() =>
+                  session.userId ? (
+                    <Redirect to="/myaccount/profile" />
+                  ) : (
+                    <SignInPage />
+                  )}
               />
               <Route
                 exact
-                path={"/myaccount/profile"}
+                path="/myaccount/sign-up"
+                key={"signUp"}
+                render={() =>
+                  session.userId ? (
+                    <Redirect to="/myaccount/profile" />
+                  ) : (
+                    <SignUpPage />
+                  )}
+              />
+              <Route
+                exact
+                path="/myaccount/profile"
                 key={"profile"}
-                component={ProfilePage}
+                render={() =>

```

Comment:
Seems like you're uh, repeating a function that outputs something to be rendered. Maybe, a...component of sorts? Tbh you can inline it on this page